authenticate-oauth/Web/Authenticate/OAuth.hs

```diff
@@ -355,12 +375,21 @@ injectOAuthToCred oa cred =
             ] cred
 
 
+-- Note that prefix is used for realm in addAuthHeader, so should fix that up,
+-- but may need an API break for that.
+addAuthBody :: BS.ByteString -> Credential -> Request -> Request
+addAuthBody _ (Credential cred) req = urlEncodedBody (filterCreds cred) req
```

Comment:
So one can use either function depending on which way you want the request signature attached to the request. Perhaps theres a better way to do this :) - I'm new to Haskell and very happy to be guided.